src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/UnitTestingIncrementalAnalyzerProvider.cs

```diff
@@ -10,11 +11,26 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting
     internal class UnitTestingIncrementalAnalyzerProvider : IIncrementalAnalyzerProvider
     {
         private readonly IUnitTestingIncrementalAnalyzerProviderImplementation _incrementalAnalyzerProvider;
+        private IIncrementalAnalyzer _analyzer;
+        private Workspace _workspace;
 
         public UnitTestingIncrementalAnalyzerProvider(IUnitTestingIncrementalAnalyzerProviderImplementation incrementalAnalyzerProvider)
             => _incrementalAnalyzerProvider = incrementalAnalyzerProvider;
 
         public IIncrementalAnalyzer CreateIncrementalAnalyzer(Workspace workspace)
-            => new UnitTestingIncrementalAnalyzer(_incrementalAnalyzerProvider.CreateIncrementalAnalyzer(workspace));
+        {
+            if (_workspace != null && _workspace != workspace)
```

Comment:
Is the current method really called twice in the same VS session?

Reply:
we should never be called for different workspaces, but this method is called on every Reanalyze call, to give back an analyzer

Reply:
I thought that this is a more adequate place to cache the analyzer, than to do it in the caller class